src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/Converters/JsonDictionaryDateTimeKeysConverter.cs

```diff
@@ -16,6 +17,8 @@ internal sealed class JsonDictionaryDateTimeKeysConverter : JsonConverter
 
         public static readonly JsonDictionaryDateTimeKeysConverter Instance = new JsonDictionaryDateTimeKeysConverter();
 
+        private readonly ConcurrentDictionary<Type, bool> _canConvertCache = new ConcurrentDictionary<Type, bool>();
```

Comment:
Isn't it better to use regular dictionary the way we use it here https://github.com/ravendb/ravendb/blob/v5.1/src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/Converters/CachingJsonConverter.cs ?

Reply:
Yes, probably better, the number of entries is small, so will give better perf.

Reply:
No, you can't, it will die on concurrent accesses. I tried locking and it is good, but so far not better than the concurrent dictionary.

Reply:
Concurrent reads are not a problem IIRC, check how this is done in the link that I have send you.

Reply:
There are no concurrent writes or writes & reads.
We create a new instance each time, because the number of times involved is very small.

Reply:
It is a switch-on-change, that would work too. My question here is: Do we want to complicate it when the performance impact is negligible?

Reply:
For the common case, for reads only, it is cheaper than concurrent dictionary, yes.